### PR TITLE
GSL2 test tolerances

### PR DESCRIPTION
**Pull request by rosswhitfield**
This reduces the tolerances for some test to allow them to pass using GSL2. See #15421

Unit tests: `PawleyFitTest`, `IqtFitMultipleTest`
Doc tests: `ConvolutionFitSequential`, `EnggCalibrate`

**To test:**
- Check that the tolerance changes aren't too drastic.
- Make sure these tests pass when using GSL2

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [ ] Does the code conform to the coding standards? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/algorithms/EnggCalibrate-v1.rst [hunk 1]
```diff
@@ -110,9 +110,9 @@ Output:
 .. testoutput:: ExampleCalib
 
    DIFA1: 0.00
-   DIFC1: 18400.19
+   DIFC1: 18400.2
    TZERO1: -2.06
    The output table has 1 row(s)
-   Parameters from the table, DIFC1: 18400.19, ZERO1: -2.06
+   Parameters from the table, DIFC1: 18400.2, ZERO1: -2.06
```
Comment (FedeMPouzols) [inline docs/source/algorithms/EnggCalibrate-v1.rst]: The changes in `EnggCalibrate` look more than acceptable. This is just a very rough/approximate calculation for the doc-test example. Feel free to tweak the values as needed.
If these are the only unit tests that need updates for GSL2 it seems that we got lucky!

File: Framework/PythonInterface/test/python/plugins/algorithms/IqtFitMultipleTest.py [hunk 2]
```diff
@@ -123,7 +123,7 @@ def _validate_group_values(self, groupWS):
         # Check Calc
         calc = sub_ws.readY(1)
         self.assertEquals(round(calc[0], 5), 1)
-        self.assertEquals(round(calc[-1],7),0.0264651)
+        self.assertEquals(round(calc[-1],6),0.026465)
         # Check Diff
```
Comment (FedeMPouzols) [inline Framework/PythonInterface/test/python/plugins/algorithms/IqtFitMultipleTest.py]: @ElliotAOram : probably you know the meaning/implications of those changes well. It looks acceptable/sensible in principle?

**Comment by rosswhitfield** [comment]
I just added the `EnggFitDIFCFromPeaks` and `EnggFitPeaks` doctests, they needed rounding to the nearest 10. @FedeMPouzols you may want to check them.

File: docs/source/algorithms/EnggCalibrate-v1.rst (hunk 1 above)
Comment (rosswhitfield) [inline docs/source/algorithms/EnggCalibrate-v1.rst]: Unfortunately not. These are just the easy tests to fix.

**Comment by FedeMPouzols** [comment]
@rosswhitfield: the changes look good. I'm really crossing fingers for the Engg system test. We never managed to get consistent results on osx with GSL1. Let's see how GSL2 behaves.

**Comment by AndreiSavici** [comment]
Doc tests are better. I still have FitPeak failing, but that is part of #15985. Also SCDCalibratePanels (does not have an issue yet). I would need to build with Paraview, otherwise I have SaveMDWorkspaceToVTK failing as well

**Comment by AndreiSavici** [comment]
The following tests FAILED:
	468 - PythonAlgorithms_EnggFitDIFCFromPeaksTest (Failed)
	469 - PythonAlgorithms_EnggFitPeaksTest (Failed)
	1163 - CrystalTest_IntegratePeakTimeSlicesTest (Failed)
	1172 - CrystalTest_OptimizeCrystalPlacementTest (Failed)
	1183 - CrystalTest_SCDCalibratePanelsTest (Failed)

**Comment by AndreiSavici** [comment]
:shipit:

**Comment by quantumsteve** [comment]
@AntonPiccardoSelg Should `SaveMDWorkspaceToVTK` be registered when ParaView is not available?

**Comment by AntonPiccardoSelg** [comment]
@quantumsteve  Good point, it uses elements of the VatesAPI, which would not get built in this case. I suppose if `MAKE_VATES` is not set, then we probably should not register this algorithm.

**Comment by AndreiSavici** [comment]
Unit tests still failing:
	1163 - CurveFittingTest_MultiDomainFunctionTest (Failed)
	1200 - CrystalTest_OptimizeCrystalPlacementTest (Failed)
Doc test failed:
EnggCalibrate-v1.rst
```
Expected:
    DIFA1: 0
    DIFC1: 18267
    TZERO1: 277
    The output table has 1 row(s)
    Parameters from the table, DIFC1: 18267, ZERO1: 277
    Output GSAS iparam file was written? True
    Number of lines of the GSAS iparam file: 36
Got:
    DIFA1: 0
    DIFC1: 18266
    TZERO1: 278
    The output table has 1 row(s)
    Parameters from the table, DIFC1: 18266, ZERO1: 278
    Output GSAS iparam file was written? True
    Number of lines of the GSAS iparam file: 36
```

**Pull request merged**